### ISSUE TITLE
nRF52 analogRead() does not work properly

### ISSUE THREAD

**Issue by bojh** [sandeepmistry/arduino-nRF5#11]
Reading analog values on analog ports does lead to completly cracy values and it seems that reading values from different port does influence one value another.

**Comment by bojh** [comment]
I have tested the new commit (eb01946) with removed "driver_nrf//hal". The issue is still there. 
What are the reasons for to remove the hal layer? .... too small advantage?

**Comment by sandeepmistry** [comment]
@bojh I think https://github.com/sandeepmistry/arduino-nRF5/commit/233eb177105157f9e6d687550baf8ca9869ed9ec makes things better, please try.

> What are the reasons for to remove the hal layer?

Licensing.

**Comment by bojh** [comment]
... not really! - In my application I have a 100k poti for 0...3,3V to one of the analog ports and I got only in the lower range ~ 0..0.643V => 0-1023 digits! a proportional value. (based on commit 233eb17)

**Comment by prjh** [comment]
I have looked up into the nordic API docu. The diver [API ](http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v11.0.0%2Findex.html)does initialize the GAIN with GAIN1_6 (1/6). Seems reasonable. I will test that setting.

**Comment by bojh** [comment]
@sandeep Proposal: add  a new global saadcGain variable, so we can ajust the gain with in combination within the `analogReference()` method e.g.:

```
/*
 * Internal Reference is at 0.6v!
 * External Reference should be between 1v and VDDANA-0.6v=2.7v
 *
 * Warning : On Arduino Zero board the input/output voltage for SAMD21G18 is 3.3 volts maximum
 */
void analogReference( eAnalogReference ulMode )
{
  switch ( ulMode ) {
    case AR_DEFAULT:
    case AR_INTERNAL:
    default:
      saadcReference = SAADC_CH_CONFIG_REFSEL_Internal;
      saadcGain           = SAADC_CH_CONFIG_GAIN_Gain1_5;	//=> 0...3,0V
      break;

    case AR_VDD4:
      saadcReference = SAADC_CH_CONFIG_REFSEL_VDD1_4;	
      saadcGain          = SAADC_CH_CONFIG_GAIN_Gain1_4;    //=> 0...VDD(3.3V)
      break;
  }
}
```
and:

```
uint32_t analogRead( uint32_t ulPin )
{
  ...
  NRF_SAADC->CH[0].CONFIG =   ((SAADC_CH_CONFIG_RESP_Bypass   << SAADC_CH_CONFIG_RESP_Pos)   & SAADC_CH_CONFIG_RESP_Msk)
                            | ((SAADC_CH_CONFIG_RESP_Bypass   << SAADC_CH_CONFIG_RESN_Pos)   & SAADC_CH_CONFIG_RESN_Msk)
                            | ((saadcGain                     << SAADC_CH_CONFIG_GAIN_Pos)   & SAADC_CH_CONFIG_GAIN_Msk)
                            | ((saadcReference                << SAADC_CH_CONFIG_REFSEL_Pos) & SAADC_CH_CONFIG_REFSEL_Msk)
                            | ((SAADC_CH_CONFIG_TACQ_3us      << SAADC_CH_CONFIG_TACQ_Pos)   & SAADC_CH_CONFIG_TACQ_Msk)
                            | ((SAADC_CH_CONFIG_MODE_SE       << SAADC_CH_CONFIG_MODE_Pos)   & SAADC_CH_CONFIG_MODE_Msk);
  NRF_SAADC->CH[0].PSELN = pin;
  NRF_SAADC->CH[0].PSELP = pin;
...

```

**Comment by sandeepmistry** [comment]
Ok, I'll try with a 10k trimpot when I get a chance. Those comments in the code above also need to be updated, they are left overs from the SAMD core.

**Comment by sandeepmistry** [comment]
@bojh can you submit a PR for this?

**Comment by bojh** [comment]
Yes, I can do that :-)

**Comment by sandeepmistry** [comment]
Fixed via #66.
